### PR TITLE
Namespace forking

### PR DESCRIPTION
**Pull request by MarinPostma**
This PR introduces a basic forking functionality. The admin API route `POST /v1/namespaces/:from/fork/:to` forks the namespace `:from` into a new namespace `:to` from the current point in time.

Forking works by creating a copy of the source database. This copy is performed from the replication log, to ensure point-in-time consistency. The forking algorithm works as follows:
- Determine the point in time at which to fork. This is done by saving the current frame_no.
- Replay the replication log from the start until the chosen point, and write that to a new database file
- Create a new namespace by placing the database file in the correct directory
- Let the recovery process take care of regenerating a replication log and replicating to bottomless

### CODE REVIEW

**Comment by psarna** [comment]
Note: for a cheap copy, we could use XFS' (and some other fs') capabilities of copy-on-write, based on https://man7.org/linux/man-pages/man2/ioctl_ficlone.2.html . Probably not trivial with our current design where a restore also takes place, it's just an interesting concept - e.g. perhaps we could CoW-copy the wallog.

**Comment by MarinPostma** [comment]
@psarna we will iterate on that design for sure

**Comment by haaawk** [comment]
Is anything blocking this PR? @psarna @MarinPostma ?

**Comment by MarinPostma** [comment]
@haaawk nope, just waiting for review

File: bottomless/src/replicator.rs [hunk 1]
```diff
@@ -967,7 +967,7 @@ impl Replicator {
         &mut self,
         generation: Uuid,
         utc_time: Option<NaiveDateTime>,
-    ) -> Result<RestoreAction> {
```
Comment (psarna) [inline bottomless/src/replicator.rs:970]: what does this bool mean? Please add a comment

Comment (psarna) [reply bottomless/src/replicator.rs:970]: Ok, now I went over the whole change and I'm even more confused - why was this parameter added at all? It looks like it's not used anywhere, you just ignore it above.

Comment (MarinPostma) [reply bottomless/src/replicator.rs:970]: it is used there, I will had a docstring
https://github.com/libsql/sqld/pull/646/files/3b76a120da75aaf2794f4f50f039a4b35b4e6c8d#diff-3035393d5fdf2fb87106000360e07ea7386f5ef923d280dd6eea3dfc9181ba39R453

**Pull request merged**